### PR TITLE
UC| Uc1

### PR DESCRIPTION
**Pull request by hkumarmk**

### CODE REVIEW

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
hmmm jorc ping is taking more time in case of overcast, as it will not be using proxy at this moment.

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
ping test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
increased ping_time to 2000 secs, but still package installation is not completed, increasing to 3000 secs and trying.

**Comment by hkumarmk** [comment]
I see last test took ~2430 secs to finish the cloud init itself.

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
First green undercloud build :)

**Comment by hkumarmk** [comment]
overcloud failed because of neutron network id resolution problem for tempest (we disabled self.instances for neutron_network), will check this.

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
Fixed the neutron_network issue in the puppet_neutron patch

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
This build should be green I hope :)

**Comment by hkumarmk** [comment]
man bad status line :(

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test overcloud

**Comment by hkumarmk** [comment]
please test undercloud

**Comment by hkumarmk** [comment]
please test me

File: Puppetfile [hunk 1]
```diff
@@ -102,8 +101,8 @@ mod 'puppetlabs/mysql',
   :ref => '2.2.x'
 
 mod 'stackforge/neutron',
-  :git => "#{base_url}/jiocloud/puppet-neutron",
-  :ref => 'origin/master'
+  :git => "#{base_url}/hkumarmk/puppet-neutron",
```
Comment (bodepd) [inline Puppetfile]: this needs to be submitted as a separate PR, if I remember correctly, it did not work

Comment (hkumarmk) [inline Puppetfile]: I fixed the issue on puppet-neutron. I will test with the other PR (wee already have a pr to test neutron change)

**Comment by hkumarmk** [comment]
please test this

File: Puppetfile [hunk 2]
```diff
@@ -78,9 +78,8 @@ mod 'stackforge/openstacklib',
   :ref => 'origin/master'
 
 mod 'stackforge/ironic',
-  :git => "#{base_url}/JioCloud/puppet-ironic",
-  :ref => 'master'
-
+  :git => "#{base_url}/hkumarmk/puppet-ironic",
```
Comment (bodepd) [inline Puppetfile]: this is a different branch than the one from the .fixtures.

Comment (bodepd) [inline Puppetfile]: this is merged into master of jiocloud now

File: build_scripts/deploy.sh [hunk 3]
```diff
@@ -42,9 +44,13 @@ while [[ ! $ip ]] ; do
   sleep 2;
 done
 
-time $timeout 1200 bash -c "while ! bash -c 'ssh -o LogLevel=Error -o ServerAliveInterval=30 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no ${ssh_user:-jenkins}@${ip} python -m jiocloud.orchestrate ping'; do sleep 5; done"
+time $timeout ${ping_timeout:-1200} bash -c "while ! bash -c 'ssh -o LogLevel=Error -o ServerAliveInterval=30 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no ${ssh_user:-jenkins}@${ip} python -m jiocloud.orchestrate ping'; do sleep 5; done"
```
Comment (bodepd) [inline build_scripts/deploy.sh]: can you add this to the README, since it's user configuration.

Comment (bodepd) [inline build_scripts/deploy.sh]: actually, I'd like to see all of the changes form this file in a separate patch as well since they can impact builds for at/prod.

File: Puppetfile (hunk 2 above)
Comment (hkumarmk) [inline Puppetfile]: Ok ill change it

File: build_scripts/deploy.sh (hunk 3 above)
Comment (hkumarmk) [inline build_scripts/deploy.sh]: Sure, ill do it

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
now nova boot also throwing "bad status line" error :( 

Creating server st1_testjenkins-puppet-rjil-gate-overcloud-112
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/var/lib/jenkins/workspace/puppet-rjil-gate-overcloud@2/venv/src/jiocloud/jiocloud/apply_resources.py", line 254, in <module>
    num_retry=args.retry)
  File "/var/lib/jenkins/workspace/puppet-rjil-gate-overcloud@2/venv/src/jiocloud/jiocloud/apply_resources.py", line 88, in create_servers
    server_id = self.create_server(userdata_file, key_name, **s)
  File "/var/lib/jenkins/workspace/puppet-rjil-gate-overcloud@2/venv/src/jiocloud/jiocloud/apply_resources.py", line 163, in create_server
    config_drive=config_drive,
  File "/var/lib/jenkins/workspace/puppet-rjil-gate-overcloud@2/venv/local/lib/python2.7/site-packages/novaclient/v2/servers.py", line 926, in create
    **boot_kwargs)
  File "/var/lib/jenkins/workspace/puppet-rjil-gate-overcloud@2/venv/local/lib/python2.7/site-packages/novaclient/v2/servers.py", line 540, in _boot
    return_raw=return_raw, **kwargs)
  File "/var/lib/jenkins/workspace/puppet-rjil-gate-overcloud@2/venv/local/lib/python2.7/site-packages/novaclient/base.py", line 161, in _create
    _resp, body = self.api.client.post(url, body=body)
  File "/var/lib/jenkins/workspace/puppet-rjil-gate-overcloud@2/venv/local/lib/python2.7/site-packages/novaclient/client.py", line 454, in post
    return self._cs_request(url, 'POST', **kwargs)
  File "/var/lib/jenkins/workspace/puppet-rjil-gate-overcloud@2/venv/local/lib/python2.7/site-packages/novaclient/client.py", line 429, in _cs_request
    resp, body = self._time_request(url, method, **kwargs)
  File "/var/lib/jenkins/workspace/puppet-rjil-gate-overcloud@2/venv/local/lib/python2.7/site-packages/novaclient/client.py", line 402, in _time_request
    resp, body = self.request(url, method, **kwargs)
  File "/var/lib/jenkins/workspace/puppet-rjil-gate-overcloud@2/venv/local/lib/python2.7/site-packages/novaclient/client.py", line 373, in request
    **kwargs)
  File "/var/lib/jenkins/workspace/puppet-rjil-gate-overcloud@2/venv/local/lib/python2.7/site-packages/requests/api.py", line 50, in request
    response = session.request(method=method, url=url, **kwargs)
  File "/var/lib/jenkins/workspace/puppet-rjil-gate-overcloud@2/venv/local/lib/python2.7/site-packages/requests/sessions.py", line 465, in request
    resp = self.send(prep, **send_kwargs)
  File "/var/lib/jenkins/workspace/puppet-rjil-gate-overcloud@2/venv/local/lib/python2.7/site-packages/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/var/lib/jenkins/workspace/puppet-rjil-gate-overcloud@2/venv/local/lib/python2.7/site-packages/requests/adapters.py", line 415, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', BadStatusLine("''",))

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
overcloud was green(puppet-rjil-gate-overcloud/119/) , but underclould failed for last test, testing again

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
Last one was actually passed - undercloud passed, overcloud failed because of some reason cronjob for maybe-upgrade.sh did not run even a single time on oc1 even though the cron job was added. testing again

**Comment by hkumarmk** [comment]
please test me

**Comment by hkumarmk** [comment]
yey!! both undercloud and overcloud passed!

**Comment by hkumarmk** [comment]
please test me

File: .fixtures.yml [hunk 4]
```diff
@@ -76,7 +78,7 @@ fixtures:
       repo: https://github.com/jiocloud/puppet-openstacklib
       ref: origin/master
     ironic:
-      repo: https://github.com/stackforge/puppet-ironic
+      repo: https://github.com/jiocloud/puppet-ironic
```
Comment (bodepd) [inline .fixtures.yml]: please make sure that you submit this change upstream

Comment (hkumarmk) [inline .fixtures.yml]: sure

Comment (hkumarmk) [inline .fixtures.yml]: oh i checked it before, I see it has been added in latest code, but could not use it here as it expected latest code
